xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py

```diff
@@ -0,0 +1,47 @@
+from typing import Tuple, Dict, Any
+
+import numpy as np
+from smqtk_classifier import ClassifyImage
+
+from xaitk_saliency.impls.gen_classifier_conf_sal.occlusion_scoring import OcclusionScoring
+from xaitk_saliency.impls.gen_image_classifier_blackbox_sal.occlusion_based import PerturbationOcclusion
+from xaitk_saliency.impls.perturb_image.sliding_window import SlidingWindow
+from xaitk_saliency.interfaces.gen_image_classifier_blackbox_sal import GenerateImageClassifierBlackboxSaliency
+
+
+class SlidingWindowStack (GenerateImageClassifierBlackboxSaliency):
+    """
+    Encapsulation of the perturbation-occlusion method using specifically
+    sliding windows and the occlusion-scoring method.
+    """
+
+    def __init__(
+        self,
+        window_size: Tuple[int, int] = (50, 50),
+        stride: Tuple[int, int] = (20, 20),
+        threads: int = 0
+    ):
+        self._po = PerturbationOcclusion(
+            perturber=SlidingWindow(
+                window_size=window_size,
+                stride=stride,
+            ),
+            generator=OcclusionScoring(),
+            threads=threads,
+        )
+
+    def _generate(self, ref_image: np.ndarray, blackbox: ClassifyImage) -> np.ndarray:
+        return self._po.generate(ref_image, blackbox)
+
+    @classmethod
+    def get_default_config(cls) -> Dict[str, Any]:
+        # Minor override following SlidingWindow impl because of list/tuple
+        # equality check differences.
```

Comment:
Can you unpack this a bit more? Under what scenarios is this `@classmethod`  pattern required?

Reply:
I thought I would be "smart" and make user of the currying that the `SlidingWindow` implementation does, but I think its both less clear as well as "wrong" (uses defaults of `SlidingWindow`s constructor, which may be different than here).